watertap/core/plugins/tests/test_solvers.py

```diff
@@ -201,25 +175,7 @@ def test_presolve_AMPL_evaluation_error_cleans_up(self, m, s):
         m.a.value = 1
 
     @pytest.mark.unit
-    def test_presolve_ipopt_error_cleans_up(self, m, s):
-        IPOPT_presolve = IPOPT._presolve
-
-        class IpoptErrorException(Exception):
-            pass
-
-        def _bad_presolve(*args, **kwargs):
-            raise IpoptErrorException
-
-        IPOPT._presolve = _bad_presolve
-        with pytest.raises(IpoptErrorException):
-            s.solve(m)
-        self._test_bounds(m)
-        assert not hasattr(s, "_scaling_cache")
-        assert _pyomo_nl_writer_log.filters == []
-        IPOPT._presolve = IPOPT_presolve
-
-    @pytest.mark.unit
-    def test_presolve_constraint_autoscale_large_jac_error_cleans_up(self, m, s):
+    def test_solve_constraint_autoscale_large_jac_error_cleans_up(self, m, s):
```

Comment:
I don't know enough about the code to have an informed opinion, but, in general, tests becoming simpler to write such as this one is usually a good indication that the refactoring makes sense :+1: .